openhtf/__init__.py

```diff
@@ -151,15 +152,18 @@ def from_uid(cls, test_uid):
       raise UnrecognizedTestUidError('Test UID %s not recognized' % test_uid)
     return test
 
-  @property
-  def uid(self):
-    """Return a unique identifier for this Test instance.
+  def get_current_or_next_uid(self):
+    if self._executor is not None:
+      return self._executor.uid
+    return self._uid
+
+  def _increment_next_uid(self):
+    """Increments the next execution's UID.
 
-    Note that this identifier must be unique across Python invokations, so the
-    station_api UID is used as a prefix to guarantee that.  Test UID's need not
-    be unique across different hosts.
+    This identifier must be unique across invocations of execute() to
+    differentiate between different runs of the same test.
     """
-    return '%s:%s' % (station_api.STATION_API.UID, id(self))
+    self._uid = '%s:%s' % (station_api.STATION_API.UID, uuid.uuid4().hex)
```

Comment:
I needed the next uid available for other reasons, but I think I can get by with it only available when executing. I'll just put my code as an early phase or part of test_start